e2e/cypress/integration/integrations/zapier/oauth_spec.js

```diff
@@ -0,0 +1,440 @@
+        cy.get('#headerInfo').click();
+        cy.get('#integrations').click();
+        cy.get('#oauthApps').click();
+
+        // # Other users should not see the apps from other users
+        cy.contains('.item-details', oauthClientID).should('not.exist');
+
+        // # Login as sysadmin
+        cy.apiAdminLogin();
+        cy.visit(testChannelUrl1);
+
+        // # Navigate to OAuthApps in integrations menu
+        cy.get('#headerInfo').click();
+        cy.get('#integrations').click();
+        cy.get('#oauthApps').click();
+
+        // * Sys admin should see the app
+        cy.contains('.item-details', oauthClientID).should('exist').within(() => {
+            // * Sys admin should see the Edit button
+            // # Click on the edit button
+            cy.findByText('Edit').should('exist').click();
+        });
+
+        // # Update description
+        cy.get('#description').type('Edited');
+
+        // # Save
+        cy.get('#saveOauthApp').click();
+
+        cy.contains('.item-details', oauthClientID).should('exist').within(() => {
+            // * Description should be editted
+            cy.findByText('TestEdited').should('exist');
+        });
+
+        // # Visit a channel
+        cy.visit(testChannelUrl1);
+
+        // # Wait for it to load
+        cy.wait(TIMEOUTS.FIVE_SEC);
+
+        cy.getCurrentChannelId().then((channelId) => {
+            const message = 'OAuth test 02';
+
+            // # Post message using OAuth credentials
+            cy.postIncomingWebhook({url: getWebhookBaseUrl() + '/postOAuthMessage', data: {channelId, message}});
+
+            // * The message should be posted
+            cy.findByText(message).should('exist');
+        });
+    });
+
+    it('MM-T650 Deauthorize Zapier', () => {
+        cy.apiLogin(user1);
+        cy.visit(testChannelUrl1);
+
+        // # Go to OAuth apps settings
+        cy.get('#headerInfo').click();
+        cy.get('#accountSettings').click();
+        cy.get('#securityButton').click();
+        cy.get('#appsEdit').click();
+
+        // * The app we created should be present
+        // # Click deauthorize
+        cy.get(`[data-app="${oauthClientID}"]`).should('exist').click();
+
+        // * The app should no longer exist
+        cy.get(`[data-app="${oauthClientID}"]`).should('not.exist');
+
+        // # Close the account settings modal
+        cy.get('#accountSettingsHeader').within(() => {
+            cy.get('button.close').click();
+        });
+
+        cy.getCurrentChannelId().then((channelId) => {
+            const message = 'OAuth test 03';
+
+            // # Post message using OAuth credentials
+            cy.postIncomingWebhook({url: getWebhookBaseUrl() + '/postOAuthMessage', data: {channelId, message}});
+
+            // * The message should not be posted
+            cy.findByText(message).should('not.exist');
+        });
+    });
+
+    it('MM-T651 Part 1 Reconnect Zapier', () => {
```

Comment:
Please change test key format to "[tm4j_id]_[step_number] [title]", so this should be "MM-T651_1 ...".

I'll update the actual test case later.